controller/src/main/java/org/jboss/as/controller/registry/ImmutableManagementResourceRegistration.java

```diff
@@ -252,4 +253,11 @@
      * @return the set of ordered child types, and and empty set if there are none
      */
     Set<String> getOrderedChildTypes();
+
+    /**
+     * Returns capability definition for this resource.
+     *
+     * @return RuntimeCapability if any registered otherwise null
+     */
+    Set<RuntimeCapability<?>> getCapabilities();
```

Comment:
I'm a bit uncomfortable about exposing the actual RuntimeCapability instances here. A capability provider's RuntimeCapability instances are really not meant to leak outside that provider's own code and the kernel.

WDYT about using AbstractCapability? That includes the metadata type stuff without including any of the service discovery and runtime API discovery stuff that shouldn't leak out. That kind of use-case distinction is the (vague) reason why I used the superclass.